rust-arroyo/src/processing/strategies/reduce.rs

```diff
@@ -14,7 +14,9 @@ use super::InvalidMessage;
 
 struct BatchState<T, TResult> {
     value: Option<TResult>,
-    accumulator: Arc<dyn Fn(TResult, T) -> TResult + Send + Sync>,
+    accumulator: Arc<
+        dyn Fn(TResult, Message<T>) -> Result<TResult, (SubmitError<T>, TResult)> + Send + Sync,
```

Comment:
I did not change the signature to borrow TResult exclusively here because I am not sure how we are using this in snuba. But I really prefer that we exclusively borrow TResult here instead of transfering ownership, this way the failure type would be easier to understand.

Reply:
I agree with you, it is probably possible to take `&mut TResult` instead